synthetics/test_schema.go

```diff
@@ -102,61 +105,65 @@ func makeTestSettingsSchema(mode schemaMode) *schema.Schema {
 		"trace":               makeTestTraceSchema(mode),
 		"port": {
 			Type:     schema.TypeInt,
-			Optional: optionalOnCreate(mode),
+			Optional: optionalOnRead(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 		},
 		"protocol": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
 			Computed: computedOnRead(mode),
 		},
 		"family": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: optionalOnRead(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 			// enumeration: "IP_FAMILY_UNSPECIFIED", "IP_FAMILY_V4", "IP_FAMILY_V6", "IP_FAMILY_DUAL"
 		},
 		"servers": {
 			Type:     schema.TypeList,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
 			Computed: computedOnRead(mode),
 			Elem: &schema.Schema{
 				Type: schema.TypeString,
 			},
 		},
 		"use_local_ip": {
 			Type:     schema.TypeBool,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
 			Computed: computedOnRead(mode),
 		},
 		"reciprocal": {
 			Type:     schema.TypeBool,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
 			Computed: computedOnRead(mode),
 		},
 		"rollup_level": {
 			Type:     schema.TypeInt,
-			Optional: optionalOnCreate(mode),
+			Optional: optionalOnRead(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 		},
 	})
 }
 
 func makeTestHostnameSchema(mode schemaMode) *schema.Schema {
-	return makeNestedObjectSchema(mode, map[string]*schema.Schema{
+	return makeNestedObjectSchemaRequired(mode, map[string]*schema.Schema{
 		"target": {
 			Type:     schema.TypeString,
-			Optional: optionalOnCreate(mode),
+			Optional: optionalOnRead(mode),
+			Required: requiredOnCreate(mode),
 			Computed: computedOnRead(mode),
 		},
 	})
 }
 
 func makeTestIPSchema(mode schemaMode) *schema.Schema {
-	return makeNestedObjectSchema(mode, map[string]*schema.Schema{
+	return makeNestedObjectSchemaOptional(mode, map[string]*schema.Schema{
 		"targets": {
 			Type:     schema.TypeList,
-			Optional: optionalOnCreate(mode),
+			Optional: true,
```

Comment:
The `targets` attribute is required on create for test type `ip`.

Reply:
Addressed:a2e73bcd0db6697b5392fdb69a572f74228bc594

Reply:
Done